### PR TITLE
perf(projects): add collapse field for unused project features

### PR DESCRIPTION
**Pull request by JoshFerge**
In large organizations with lots of projects, these feature flag computations get expensive and take up to ~350ms each.

This PR aims to remove unused feature flags from the projects serialized if a new collapse field, `unusedFeatures` is set.

Related: https://github.com/getsentry/sentry/issues/65947

Frontend PR to follow

### CODE REVIEW

File: src/sentry/api/serializers/models/project.py [hunk 1]
```diff
@@ -60,6 +60,24 @@
 _PROJECT_SCOPE_PREFIX = "projects:"
 
 LATEST_DEPLOYS_KEY: Final = "latestDeploys"
+UNUSED_ON_FRONTEND_FEATURES: Final = "unusedFeatures"
+
+
+# These features are not used on the frontend,
+# and add a lot of latency ~100-300ms per flag for large organizations
+# so we exclude them from the response if the unusedFeatures collapse parameter is set
+PROJECT_FEATURES_NOT_USED_ON_FRONTEND = {
```
Comment (scefali) [inline src/sentry/api/serializers/models/project.py:69]: Maybe we could register certain features as being front-end only? This is probably OK for now but keeping this in sync with features long term isn't great

Comment (JoshFerge) [reply src/sentry/api/serializers/models/project.py:69]: The intention is to move away from loading all projects at app init. -- once we do that, we can likely remove this.

**Review by scefali — APPROVED** [review]

File: src/sentry/api/serializers/models/project.py (hunk 1 above)
Comment (JoshFerge) [reply src/sentry/api/serializers/models/project.py:69]: also note that this only excludes ones defined in this list -- new project flags will by default be in the response.

Comment (wedamija) [reply src/sentry/api/serializers/models/project.py:69]: I think there's a case to be made for only including features in the frontend response when they're actually necessary, and including this in the feature flag registration process might make sense. We have hundreds of features at the moment, and presumably the frontend doesn't really use all of these.

Doesn't have to be in this pr, but I think it's worth considering

**Review by wedamija — APPROVED** [review]

File: src/sentry/api/serializers/models/project.py (hunk 1 above)
Comment (JoshFerge) [reply src/sentry/api/serializers/models/project.py:69]: for sure, definitely worth considering as we're making more changes to feature handling going forward

**Pull request merged**